tools/dataflow-throttling/README.md

```diff
@@ -0,0 +1,116 @@
+# Client-side Throttling in Dataflow
+
+### Introduction
+When Dataflow pipeline is trying to get the response for each incoming event from third-party API’s there is a chance of the request to fail in getting response for some reasons like network issues, limits on API calls, etc. At times, client-side throttling can act as a way to limit the number of requests coming from dataflow by either postponing or dropping excessive requests. It can also serve as a fail-safe mechanism that prevents the loss of payload of requests throttled by the server.
```

Comment:
When Dataflow pipeline is trying to get a response for an incoming event from third-party API there is a chance of the request to fail in getting response, for example because of network issues, limits on API calls, etc. Client-side throttling can act as a way to limit the number of requests coming from dataflow by either postponing or dropping excessive requests. It can also serve as a fail-safe mechanism that prevents the loss of payload of requests throttled by the server by storing failed requests in DLQ.